[CI] SetSingleNodeAllocateStepTests testPerformActionSomeShardsOnlyOnNewNodesButNewNodesInvalidAttrs failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6135 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/kc76r2buewrki)
- [elasticsearch-pull-request #54458 / part-2](https://gradle-enterprise.elastic.co/s/sr2u643xaeea6)
- [elasticsearch-periodic #6104 / openjdk21_checkpart2_java-fips-matrix](https://gradle-enterprise.elastic.co/s/74sg3ibvdsjls)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:core:test" --tests "org.elasticsearch.xpack.core.ilm.SetSingleNodeAllocateStepTests.testPerformActionSomeShardsOnlyOnNewNodesButNewNodesInvalidAttrs" -Dtests.seed=6DD3B21F36892442 -Dtests.locale=pt-GW -Dtests.timezone=Asia/Brunei -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.core.ilm.SetSingleNodeAllocateStepTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testPerformActionSomeShardsOnlyOnNewNodesButNewNodesInvalidAttrs),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unexpected
```

**Issue Reasons:**
- [main] 3 failures in test testPerformActionSomeShardsOnlyOnNewNodesButNewNodesInvalidAttrs (1.1% fail rate in 274 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -341,9 +341,6 @@ tests:
 - class: org.elasticsearch.ingest.geoip.FullClusterRestartIT
   method: testGeoIpSystemFeaturesMigration {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/121115
-- class: org.elasticsearch.xpack.core.ilm.SetSingleNodeAllocateStepTests
-  method: testPerformActionSomeShardsOnlyOnNewNodesButNewNodesInvalidAttrs
-  issue: https://github.com/elastic/elasticsearch/issues/121495
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only}
   issue: https://github.com/elastic/elasticsearch/issues/121412

